### PR TITLE
fix: 8329: Resolve fabmenu open multiple time

### PR DESCRIPTION
**Pull request by dharmik**
### Fixed Issues

$ [8329](https://github.com/Expensify/App/issues/8329)

### Tests

tests to something like this -

1 Login to New expensify
2 In RHN, tap on the green +
Double tap on new chat
3 Verify that the new chat page opens

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

tests to something like this -

1 Login to New expensify
2 In RHN, tap on the green +
Double tap on new chat
3 Verify that the new chat page opens

- [x] Verify that no errors appear in the JS console

### Video

iOS

https://user-images.githubusercontent.com/1510175/163661634-b94814f7-593e-4f2a-a765-d71e958ec7e5.mov

android

https://user-images.githubusercontent.com/1510175/163661667-e2886517-131a-44e4-bf9c-e05596f19d9a.mp4

Macos

https://user-images.githubusercontent.com/1510175/164064363-a4896dcd-bf6d-4e83-9f11-3a72da6da420.mov

MWeb

https://user-images.githubusercontent.com/1510175/164297929-92f53655-2880-4713-aca3-a3d17986bf00.mov

Web

https://user-images.githubusercontent.com/1510175/164299216-2df0951e-6024-46b2-a009-ed10bf8f9cbd.mov

### CODE REVIEW

**Comment by dharmik** [comment]
@rushatgabhane PR updated.

**Comment by rushatgabhane** [comment]
@dharmik please add screenshots for all platforms

**Review by tgolen — CHANGES_REQUESTED** [review]

File: src/pages/home/sidebar/SidebarScreen.js [hunk 1]
```diff
@@ -58,14 +59,23 @@ class SidebarScreen extends Component {
...
 
         const routes = lodashGet(this.props.navigation.getState(), 'routes', []);
-        WelcomeAction.show({routes, toggleCreateMenu: this.toggleCreateMenu});
+        WelcomeAction.show({routes, hideCreateMenu: this.hideCreateMenu});
     }
 
     /**
      * Method called when a Create Menu item is selected.
      */
     onCreateMenuItemSelected() {
```
[the remark this answers is not in the document]
Comment (rushatgabhane) [reply src/pages/home/sidebar/SidebarScreen.js:68]: @dharmik, `onCreateMenuItemSelected()` just calls `hideCreateMenu()`.

So you need to 

1. Remove `onCreateMenuItemSelected()`, and
2. Replace all calls to it with `hideCreateMenu()`

Let me know if you have any questions.

**Comment by dharmik** [comment]
@rushatgabhane I updated videos in PR description is that okay?

File: src/pages/home/sidebar/SidebarScreen.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dharmik) [reply src/pages/home/sidebar/SidebarScreen.js]: @rushatgabhane Resolved

[the remark this answers is not in the document]
Comment (dharmik) [reply src/pages/home/sidebar/SidebarScreen.js]: @rushatgabhane @tgolen Can you please share your opinion on it? https://github.com/Expensify/App/pull/8692#discussion_r854146319

File: src/libs/actions/WelcomeActions.js [hunk 2]
```diff
@@ -75,7 +75,7 @@ function show({routes, toggleCreateMenu}) {
         // It's also possible that we already have a workspace policy. In either case we will not toggle the menu but do still want to set the NVP in this case
```
[the remark this answers is not in the document]
Comment (rushatgabhane) [reply src/libs/actions/WelcomeActions.js:75]: @dharmik ig you missed this

File: src/pages/home/sidebar/SidebarScreen.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rushatgabhane) [reply src/pages/home/sidebar/SidebarScreen.js]: > Can you please share your opinion on it? https://github.com/Expensify/App/pull/8692#discussion_r854146319

@dharmik I'm sorry, I don't understand. I thought this conversation has been resolved. 
Only action required by you is to address https://github.com/Expensify/App/pull/8692#discussion_r853460740

File: src/libs/actions/WelcomeActions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dharmik) [reply src/libs/actions/WelcomeActions.js]: @rushatgabhane PR updated.

**Comment by dharmik** [comment]
@rushatgabhane Can you please recheck this PR & let me know if I miss something?

**Review by rushatgabhane — APPROVED** [review]
@tgolen Looks good!
<details>
<summary><h4> #### PR Reviewer Checklist</h4></summary>

- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [ ] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

**Comment by rushatgabhane** [comment]
@dharmik please address [this comment](https://github.com/Expensify/App/pull/8692#issuecomment-1103117551). Thanks so much!

**Comment by dharmik** [comment]
@rushatgabhane 

https://github.com/Expensify/App/pull/8692

I added all videos.

**Comment by rushatgabhane** [comment]
@dharmik the tests aren't clear, so please address that too. My comment [here](https://github.com/Expensify/App/pull/8692#issuecomment-1106853652) was a suggestion for that

**Comment by dharmik** [comment]
if in those video testes are not clear so in screenshot how clear?

Appreciate your suggestion 

@rushatgabhane

**Comment by rushatgabhane** [comment]
Ohh I see. Forget the video completely. As per the suggestions [here](https://github.com/Expensify/App/pull/8692#issuecomment-1103117551)  update these tests in the PR description 👇

Screenshot_20220423-230513

**Comment by dharmik** [comment]
PR description updated @rushatgabhane

**Comment by rushatgabhane** [comment]
Btw, PR is [approved](https://github.com/Expensify/App/pull/8692#pullrequestreview-950568576)!

**Review by tgolen — APPROVED** [review]

**Comment by marcaaron** [comment]
Sorry if I'm missing some context but doesn't this PR break the "automatic" opening on the create menu for a first time user sign in since we are only "hiding" it now?

I find it kind of weird that we are calling `WelcomeAction.show()` but it doesn't "show" anything and either navigates somewhere or calls a `hide()` method.

**Comment by marcaaron** [comment]
https://github.com/Expensify/App/blob/db6fab75722e833ca6e54f634e8a6b6f3c1cae79/src/libs/actions/WelcomeActions.js#L43-L81

Anyone can tell us where in this function we "show" something?

**Comment by marcaaron** [comment]
Thinking maybe it should have been `showCreateMenu()` and we just got mixed up because it wasn't clear if "toggle" meant "hide" or "show". But it means "show" right?

**Comment by marcaaron** [comment]
(**context:** While looking into the issue [here](https://github.com/Expensify/App/issues/5296) realized the code no longer makes sense since we're not showing anything)

**Comment by tgolen** [comment]
Those are pretty good questions. I actually think there is quite a bit about the welcome actions code that is an anti-pattern (like passing callback functions into the action file, the use of the setTimeout(), etc.). I would say there aren't really any good answers, and I would be in favor of just doing a bit of an over-haul and cleanup to this code.

**Comment by marcaaron** [comment]
I agree, but this PR still broke existing behavior. We are calling a function that doesn't show anything and won't hide something because nothing is shown in the first place.

**Comment by tgolen** [comment]
Yeah, so in that case, it's definitely a regression that needs fixed regardless. You are right.

**Pull request merged**